OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo

```diff
@@ -192,6 +194,79 @@ algorithm
   end if;
 end removeVerySimpleEquations;
 
+public function fixAliasVarsVariablity
+" This is a workaround for #5673
+  This module traverses all alias variables and double-checks if the referenced alias variables variablitly
+  is parameter or not and changes variablilty of alias variable to BackendDAE.PARAM()."
+  input BackendDAE.BackendDAE inDAE;
+  output BackendDAE.BackendDAE outDAE;
+protected
+  BackendDAE.Variables aliasVars,systvars,globalKnownVars;
+  DAE.Exp binding;
+  DAE.ComponentRef cr;
+  Boolean param;
+  BackendDAE.Shared shared;
+  list<BackendDAE.Var> referencevar,aliasVarList = {},knownVarList={};
+algorithm
+  aliasVars := BackendDAEUtil.getAliasVars(inDAE);
+  systvars := getVarsFromEqSystem(inDAE.eqs);
+  //BackendDump.dumpVariables(aliasVars,"aliasVariable-Actual");
+  for var in BackendVariable.varList(aliasVars) loop
+    binding := BackendVariable.varBindExp(var);
+    {cr} := Expression.getAllCrefs(binding);
+    // look up for the reference variable variablity
+    if BackendDAEUtil.containAnyVar({cr}, systvars) then  //check in EqSystem.OrderedVars
```

Comment:
Can you create a new function that fetches a given variable from the ordered vars directly, without the need of `getVarsFromEqSystem`? And the same for global known vars.